backends/kv/inferschema.go

```diff
@@ -83,15 +85,24 @@ func schemaFromKeys(keyField string, rowSet []map[string]interface{}) (v3ioutils
 	columnCanBeFullKey := make(map[string]bool)
 	columnCanBePrimaryKey := make(map[string]bool)
 	columnCanBeSortingKey := make(map[string]bool)
+	columnCanBeHashedPrimaryKey := make(map[string]bool)
 
 	for _, row := range rowSet {
 		keyValue := row["__name"].(string)
 		var primaryKeyValue string
 		var sortingKeyValue string
+		var hashedPrimaryKeyValue string
 		indexOfDot := strings.Index(keyValue, ".")
 		if indexOfDot >= 0 && indexOfDot < len(keyValue)-1 {
 			sortingKeyValue = keyValue[indexOfDot+1:]
-			primaryKeyValue = keyValue[:indexOfDot]
+			potentialPrimaryKey := keyValue[:indexOfDot]
+
+			if hashedBucketFormat.MatchString(potentialPrimaryKey) {
```

Comment:
We should probably not break the case where the keys match the regex by mistake, not because of hashing. The user is still allowed to use `tal_1` and `tal_2` as keys. So a test like this should probably pass (but currently doesn't):
```
func (suite *InferSchemaTestSuite) TestInferSchemaWithHashedPrimaryKey() {
	expectedFields := []v3ioutils.OldSchemaField{
		{Name: "name", Type: "string", Nullable: false},
		{Name: "animal", Type: "string", Nullable: false},
		{Name: "age", Type: "long", Nullable: true},
	}

	keyField := ""
	rowSet := []map[string]interface{}{
		{"__name": "rocky_2.dog", "name": "rocky_2", "animal": "dog", "age": 2},
		{"__name": "mocha_2.dog", "name": "mocha_2", "animal": "dog", "age": 3},
		{"__name": "mocha_4.cat", "name": "mocha_4", "animal": "cat", "age": 9},
	}
	schema, err := schemaFromKeys(keyField, rowSet)
	suite.Require().NoError(err)
	concreteSchema := schema.(*v3ioutils.OldV3ioSchema)
	suite.Require().Equal("name", concreteSchema.Key)
	suite.Require().Equal("animal", concreteSchema.SortingKey)
	suite.Require().ElementsMatch(expectedFields, concreteSchema.Fields)
	suite.Require().Equal(0, concreteSchema.HashingBucketNum)
}
```